### ISSUE TITLE
Mouse wheel direction of overhead air vents is inverted

### ISSUE THREAD

**Issue by wkitty42** [Juanvvc/c172p-detailed#613]
On the control panel, scrolling the mouse wheel forward decrements the value of what we altering and pulling the mouse wheel back towards you increments the value.

However, for the overhead air vents in the top left and right corners of the windshield the direction of the mouse wheel is inverted.

We should push the mouse wheel away from us to close the vents just like we push the vent to close it.

**Comment by onox** [comment]
Dragging direction is OK, but mouse wheel scrolling of the overhead air vents is not.

@wkitty42 Please end phrases with a single dot instead of three.

**Comment by wkitty42** [comment]
this problem also exists with the key switch and the instrument brightness knob... should i make a new issue for each of these or no??

**Comment by onox** [comment]
No.

**Comment by wkitty42** [comment]
ok :smile_cat:

**Comment by onox** [comment]
What's wrong with the magnetos? Drag and mouse wheel scrolling seems OK.

**Comment by wkitty42** [comment]
@onox : sorry... those three dots is from 30 years of muscle memory...

**Comment by onox** [comment]
Instrument brightness knob is also OK. They're horizontal, not vertical like the throttle/mixture/cabinheat/cabineair sliders.

**Comment by wkitty42** [comment]
maybe i need to make another pull? rolling the mouse wheel forward should turn the key off. off is down (aka decrement). pushing the top of the wheel forward should decrement.

i *always* roll forward to decrease and pull back to increase... no matter if they are twisty knobs or horizontal/vertical sliders or in/out sliders like the mixture and throttle...

**Comment by onox** [comment]
> pushing the top of the wheel forward should decrement.

I don't think so. Pushing mouse wheel forward should turn the key clockwise. Which it does.

**Comment by gilbertohasnofb** [comment]
Yep, pushing something forwards == increase IMO

**Comment by onox** [comment]
On the other hand: drag direction of flap should be vertical instead of horizontal.

**Comment by gilbertohasnofb** [comment]
@onox Maybe we should open a new issue for the flap as well, so this doesn't get forgotten in this comment of yours.

**Comment by onox** [comment]
Sure.

**Comment by gilbertohasnofb** [comment]
Ok, see: https://github.com/Juanvvc/c172p-detailed/issues/616

**Comment by wkitty42** [comment]
On 12/11/2015 03:30 PM, onox wrote:
>     pushing the top of the wheel forward should decrement.
>
> I don't think so. Pushing mouse wheel forward should turn the key clockwise.
> Which it does.

sorry, that's backwards to me...

i should also rephrase...

pushing in, turning left, lifting up should have the mouse wheel being pushed...

pulling out, turning right, pressing down should have the mouse wheel being 
pulled...


the thing is consistency... not really the increasing or decreasing of the 
value... the throttle and mixture are examples of increasing the value when we 
push in...

to turn the key off, you turn left... to turn the lights down, you turn left... 
to change the web bug position CCW, you turn left... it doesn't matter if the 
indicator is vertical or horizontal...

am i seeing this because i'm right handed? i'm really only after consistency... 
it always catches me out on some knobs or sliders when i push in expecting one 
thing and it does the opposite... kinda like having the steering wheel of you 
car having to be turned right to actually turn left...

**Comment by gilbertohasnofb** [comment]
> am i seeing this because i'm right handed? i'm really only after consistency...

I am also right handed :wink: Where do you see the inconsistency in our plane? Inconsistency means that in some cases it behaves one way and others it behaves differently, so which two switches/levers behave in opposite way? I think we have all of them consistent now; when it's a lever, that's easy: rolling forward the wheel == pushing the lever forward. When it's a rotating switch, forward == clockwise, or incrementing the value.

I just tested every single knob, switch and lever and they all behave as I wrote above with one single exception: the flaps are inverted. I will add a comment about it to https://github.com/Juanvvc/c172p-detailed/issues/616

But about the rest, I think this issue is not really an issue :wink:

**Comment by dany93** [comment]
I already wrote this comment about the scroll wheel direction https://github.com/Juanvvc/c172p-detailed/issues/444

**Comment by gilbertohasnofb** [comment]
Let's suppose that this is really an issue in which 50% percent of people will have one opinion and 50% will have another. If that's the case (which I really don't know if it is since I only saw you and @wlbragg saying the prefer the opposite behaviour) then as long as the behaviour of scrolling is consistent -- which it is -- then it's a matter of picking one option over another. Then given that the plane has this present behaviour for some time already and AFAIK there weren't any complaints about it the forum or mailing list, then I'd just say that we stick with what we have.

**Comment by dany93** [comment]
This aircraft initially had the opposite behaviour for the scroll wheel on the knobs. There were no complaints. What I don't understand why you changed it.

**Comment by gilbertohasnofb** [comment]
> What I don't understand why you changed it.

I didn't, Dany, it was someone else -- but I do prefer the current behaviour much more. I am only making a point of why I think reverting back would be a bad decision. And I am also making a point on why I think the current behaviour is good (forward = increase seems fairly logical) as well as showing that we have consistency.

**Comment by dany93** [comment]
"You" was not for you gilberto, personally.
But not easy I admit, we can find both, depending on FG aircraft...

**Comment by onox** [comment]
If you want the opposite behavior, see https://github.com/Juanvvc/c172p-detailed/issues/444#issuecomment-133879809

**Comment by wkitty42** [comment]
On 12/11/2015 06:20 PM, Gilberto Agostinho wrote:
>     am i seeing this because i'm right handed? i'm really only after consistency...
>
> I am also right handed :wink: Where do you see the inconsistency in our plane?
> Inconsistency means that in some cases it behaves one way and others it behaves
> differently, so which two switches/levers behave in opposite way? I think we
> have all of them consistent now; when it's a lever, that's easy: rolling forward
> the wheel == pushing the lever forward. When it's a rotating switch, forward ==
> clockwise, or incrementing the value.

i was kinda incorrect earlier when i was trying to associate pushing with 
decreasing... i thought that was proper but then the throttle and mixture don't 
do that... they increase when pushed in... now trying to explain how it is 
backwards to me is confusing me :(

my mouse is canted slightly to the left most of the time... kinda like a person 
with pigeon toes has their toes pointing inwards toward each other... that puts 
pushing the mouse wheel in roughly the same orientation as turning 
counter-clockwise (or anti-clockwise as some say)... when i want to turn a knob 
to the left, i naturally push my mouse wheel forward...

> I just tested every single knob, switch and lever and they all behave as I wrote
> above with one single exception: the flaps are inverted. I will add a comment
> about it to #616 <https://github.com/Juanvvc/c172p-detailed/issues/616>
>
> But about the rest, I think this issue is not really an issue :wink:

it is, i guess, more of an aggravation... every time i go to turn the lights 
down, i have to roll the wheel the opposite direction instead of the natural 
pushing the wheel forward to turn the lights down... the radio volume knobs are 
also backwards to me as are the radio knobs to change the frequency... pushing 
the wheel doesn't reduce the frequency... the same with the AP altitude 
setting... pushing increases the altitude setting...

look at it like this... you have a volume knob in front of you... you put your 
right hand's index finger tip on the top of that knob... now twist your hand so 
that your index finger is pointing to the left and your finger tip is still on 
the top of the knob... now, turn the volume down... what did you do with your 
finger to do that? you pushed the top of the knob to the left... pushed... just 
like pushing the mouse wheel should turn the volume down...

**Comment by wkitty42** [comment]
On 12/11/2015 06:25 PM, dany93 wrote:
> I already wrote this comment about the scroll wheel direction #444

thank you! you sure did and what you say there seems to be the same as what i'm 
trying to say here...

**Comment by wkitty42** [comment]
On 12/11/2015 06:29 PM, Gilberto Agostinho wrote:
> Let's suppose that this is really an issue in which 50% percent of people will
> have one opinion and 50% will have another. If that's the case (which I really
> don't know if it is since I only saw you and @wlbragg
> <https://github.com/wlbragg> saying the prefer the opposite behaviour) then as
> long as the behaviour of scrolling is consistent -- which it is -- then it's a

yes, it is consistently backward :wink:

> matter of picking one option over another. Then given that the plane has this
> present behaviour for some time already and AFAIK there weren't any complaints
> about it the forum or mailing list, then I'd just say that we stick with what we
> have.

lack of published complaints... true... however, some folks hope and watch until 
things just finally get to be aggravating to them before they voice their 
complaint...

this is at least the second or third issue that has been filed about the mouse 
wheel rotation action being backwards... i wouldn't even say anything if ""you 
guys"" hadn't pointed out that rolling the mouse wheel works instead of trying 
to click and/or drag things to adjust the knobs... now that i know about the 
mouse wheel, it is one of my most used because i don't have to keep moving the 
mouse all over the place and picking it up to get it back in place for more 
movement in one direction or another... the heel of my right hand stays planted 
on the edge of the table and i move the mouse with my fingers... my arm doesn't 
move at all... my hand pivots left and right and my fingers push and pull... 
many folks put their whole arm into moving the mouse around but i don't and 
never have... if i have moved the mouse as far to the right as i can, i pick it 
up with my fingers and move it back to the center of my area so i can keep 
moving it right if i have to... i don't know if this helps explain it or not :cry:

**Comment by wkitty42** [comment]
On 12/11/2015 07:05 PM, onox wrote:
> If you want the opposite behavior, see #444 (comment)
> <https://github.com/Juanvvc/c172p-detailed/issues/444#issuecomment-133879809>

but won't that flip the things backwards that are working properly now??

eg: throttle and mixture would be pushing the wheel to pull them out

**Comment by onox** [comment]
It probably will. The problem is that some of the controls always function the same: you scroll the mouse wheel forward to move the knob of the throttle/mixture away from you (towards the engine), but some other controls do not: these horizontal knobs should either turn clockwise or counterclockwise depending on who you ask. But you cannot make a distinction between these two.
